File: 100 Day Coding Sprint/Day 1/distinctk.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

class Main {
    public static String fn(List<String> lt, int k, int n){
        HashMap<String, Integer> map = new HashMap<>();
        List<String> order = new ArrayList<>();
        List<String> result = new ArrayList<>();

        for(String c: lt){
            map.put(c,map.getOrDefault(c,0)+1);
            if(map.get(c) == 1){
                order.add(c);
            }
        }

        for(String c: order){
            if(map.get(c)==1){
                result.add(c);
            }
        }

        if(result.size()< k){
            return "-1";
        }else{
            return result.get(k-1);
        }
    }

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();
        
        List <String> lt = new ArrayList<>(); 
        for(int i = 0; i < n; i++){
            String data = sc.nextLine();
            lt.add(data);
        }

        int k = sc.nextInt();

        System.out.println(fn(lt,k,n));
    }
}

File: 100 Day Coding Sprint/Day 1/distinctk.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <bits/stdc++.h>
#include <algorithm>
using namespace std;

string fn(vector<string> charArray, int k, int n)
{
    unordered_map<string, int> map;
    vector<string> result;
    vector<string> order;
    for (string c : charArray)
    {
        map[c]++;
        if (map[c] == 1)
        {
            order.push_back(c);
        }
    }
    for (string s : order)
    {
        if (map[s] == 1)
        {
            result.push_back(s);
        }
    }
    if (result.size() < k)
    {
        return "-1";
    }
    else
    {
        return result[k - 1];
    }
}
int main()
{
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    int n;
    cin >> n;

    vector<string> charArray;
    for (int i = 0; i < n; i++)
    {
        string data;
        cin >> data;
        charArray.push_back(data);
    }

    int k;
    cin >> k;

    cout << fn(charArray, k, n);

    return 0;
}